lib/internal/modules/cjs/loader.js

```diff
@@ -1004,8 +1004,18 @@ Module._resolveFilename = function(request, parent, isMain, options) {
     paths = Module._resolveLookupPaths(request, parent);
   }
 
-  if (parent && parent.filename) {
-    const filename = trySelf(parent.filename, isMain, request);
+  if (parent && (
+    parent.id === 'internal/preload' ||
+    parent.id === '<repl>' ||
+    parent.filename
+  )) {
+    let base, filename;
+    try {
+      base = parent.filename || process.cwd() + path.sep;
+    } catch {}
```

Comment:
I'm not sure if try-catch still causes a minor perf regression on hot code paths, but it might be worth separating it out from the standard case to be careful for performance:

```suggestion
  let base, filename;
  if (parent && (parent.filename || parent.id === '<repl>') {
    base = parent.filename || process.cwd() + path.sep;
  }
  else if (parent && parent.id === 'internal/preload') {
    try {
      base = process.cwd() + path.sep
    }
    catch {}
  }
```

Reply:
It should not cause major performance regressions but I agree on refactoring this part just to be safe.